action.yml

```diff
@@ -139,6 +149,28 @@ runs:
         echo -n "${HOST_KEY_ALGOS//$'\n'/,}" >~/.ssh/host_key_types.sshupload
         echo "Common host key types:"
         cat ~/.ssh/host_key_types.sshupload
+    - name: "Delete files before upload"
+      env:
+        # We add the parameters as environment variables, so they can be properly quoted in the shell code below.
+        HOST: ${{ inputs.host }}
+        USERNAME: ${{ inputs.username }}
+        SOURCE: ${{ inputs.source }}
+        TARGET: ${{ inputs.target }}
+        DELETE_BEFORE_UPLOAD: ${{ inputs.delete_before_upload }}
+        PORT: ${{ inputs.port }}
+        KNOWN_HOSTS: ${{ inputs.known_hosts }}
+      if: inputs.delete_before_upload == 'yes'
+      shell: bash
+      run: |
+        set -e
+        echo -e "\e[32mDeleting files in the target before upload...\e[0m"
+        cat <<<"${KNOWN_HOSTS}" >~/.ssh/known_hosts.sshupload
+        echo "SOURCE is set to: ${SOURCE}"
+        FILES=$(echo $SOURCE)
+        echo "FILES is set to: ${FILES}"
+        COMMAND='set -e; cd '${TARGET}' && for file in '${FILES}'; do echo "file: $file" && rm -v $(basename $file) || true; done'
```

Comment:
Is ${TARGET} going to be properly replaced this way?

Reply:
> Is ${TARGET} going to be properly replaced this way?

Yes, since it's outside of single quoted string, just like ${FILES}:

Running remote command: set -e; cd /*** && for file in ***data/1 ***data/11 ***data/14 ***data/16 ***data/18 ***data/19; do echo "file: $file" && rm -v $(basename $file) || true; done
file: ***data/1
removed '1'
file: ***data/11
removed '11'
file: ***data/14
removed '14'
file: ***data/16
removed '16'
file: ***data/18
removed '18'
file: ***data/19
removed '19'